unidist/core/backends/mpi/core/controller/common.py

```diff
@@ -155,15 +189,18 @@ def request_worker_data(data_id):
     )
 
     # Blocking get
-    data = communication.recv_complex_data(mpi_state.comm, owner_rank)
+    complex_data = get_data(mpi_state.comm, owner_rank)
+    if data_id.base_data_id() != complex_data["id"]:
+        raise ValueError("Unexpected data_id!")
```

Comment:
When can we get into this branch?

Reply:
Just for security purposes in case our message pipeline is broken. This check has already been before and has just replaced here from "get" function in api.py.

Reply:
ok, let's leave it